MapboxCoreNavigation/RouteController.swift

```diff
@@ -600,6 +583,27 @@ extension RouteController: CLLocationManagerDelegate {
         checkForFasterRoute(from: location)
     }
     
+    func updateRouteLegProgress(for location: CLLocation) {
+        guard routeProgress.currentLegProgress.upComingStep?.maneuverType == .arrive else { return }
+        
+        let polyline = Polyline(routeProgress.currentLegProgress.currentStep.coordinates!)
+        let userSnapToStepDistanceFromManeuver = polyline.distance(from: location.coordinate)
+        let secondsToEndOfStep = userSnapToStepDistanceFromManeuver / location.speed
+        guard secondsToEndOfStep < RouteControllerDurationRemainingWaypointArrival else { return }
```

Comment:
This guard statement should become an if statement, and everything below should be factored out into a helper method. Use guard statements only for edge cases. As it is, you have to read carefully to spot this line and know that the stuff below here only runs when arriving.